### ISSUE TITLE
After update PHPStorm to 2022.1 plugin crash on startup IDE

### ISSUE THREAD

**Issue by izac1** [kalessil/phpinspectionsea#1824]
| Plugin         | Php Inspections (EA Ultimate) latest |
| Language level | PHP 8.1 |

```
com.intellij.diagnostic.PluginException: Short name 'AmbiguousMethodsCallsInArrayMappingInspection' is not unique
class 'com.kalessil.phpStorm.phpInspectionsEA.inspectors.codeStyle.AmbiguousMethodsCallsInArrayMappingInspector' in PluginDescriptor(name=Php Inspections (EA Extended), id=com.kalessil.phpStorm.phpInspectionsEA, descriptorPath=plugin.xml, path=~/.local/share/JetBrains/Toolbox/apps/PhpStorm/ch-0/221.5080.224.plugins/PhpInspectionsEA.jar, version=4.0.7.1, package=null, isBundled=false)
and class 'com.kalessil.phpStorm.phpInspectionsEA.inspectors.codeStyle.AmbiguousMethodsCallsInArrayMappingInspector' in PluginDescriptor(name=Php Inspections (EA Ultimate), id=com.kalessil.phpStorm.phpInspectionsUltimate-mp, descriptorPath=plugin.xml, path=~/.local/share/JetBrains/Toolbox/apps/PhpStorm/ch-0/221.5080.224.plugins/PhpInspectionsUltimate-mp.jar, version=2021.5, package=null, isBundled=false)
conflict

com.intellij.diagnostic.PluginException: Short name 'ThrowRawExceptionInspection' is not unique
class 'com.kalessil.phpStorm.phpInspectionsEA.inspectors.apiUsage.ThrowRawExceptionInspector' in PluginDescriptor(name=Php Inspections (EA Extended), id=com.kalessil.phpStorm.phpInspectionsEA, descriptorPath=plugin.xml, path=~/.local/share/JetBrains/Toolbox/apps/PhpStorm/ch-0/221.5080.224.plugins/PhpInspectionsEA.jar, version=4.0.7.1, package=null, isBundled=false)
and class 'com.kalessil.phpStorm.phpInspectionsEA.inspectors.apiUsage.ThrowRawExceptionInspector' in PluginDescriptor(name=Php Inspections (EA Ultimate), id=com.kalessil.phpStorm.phpInspectionsUltimate-mp, descriptorPath=plugin.xml, path=~/.local/share/JetBrains/Toolbox/apps/PhpStorm/ch-0/221.5080.224.plugins/PhpInspectionsUltimate-mp.jar, version=2021.5, package=null, isBundled=false)
conflict



com.intellij.diagnostic.PluginException: Short name 'DuplicateArrayKeysInspection' is not unique
class 'com.kalessil.phpStorm.phpInspectionsEA.inspectors.semanticalAnalysis.DuplicateArrayKeysInspector' in PluginDescriptor(name=Php Inspections (EA Extended), id=com.kalessil.phpStorm.phpInspectionsEA, descriptorPath=plugin.xml, path=~/.local/share/JetBrains/Toolbox/apps/PhpStorm/ch-0/221.5080.224.plugins/PhpInspectionsEA.jar, version=4.0.7.1, package=null, isBundled=false)
and class 'com.kalessil.phpStorm.phpInspectionsEA.inspectors.semanticalAnalysis.DuplicateArrayKeysInspector' in PluginDescriptor(name=Php Inspections (EA Ultimate), id=com.kalessil.phpStorm.phpInspectionsUltimate-mp, descriptorPath=plugin.xml, path=~/.local/share/JetBrains/Toolbox/apps/PhpStorm/ch-0/221.5080.224.plugins/PhpInspectionsUltimate-mp.jar, version=2021.5, package=null, isBundled=false)
conflict

com.intellij.diagnostic.PluginException: Short name 'UselessUnsetInspection' is not unique
class 'com.kalessil.phpStorm.phpInspectionsEA.inspectors.apiUsage.UselessUnsetInspector' in PluginDescriptor(name=Php Inspections (EA Extended), id=com.kalessil.phpStorm.phpInspectionsEA, descriptorPath=plugin.xml, path=~/.local/share/JetBrains/Toolbox/apps/PhpStorm/ch-0/221.5080.224.plugins/PhpInspectionsEA.jar, version=4.0.7.1, package=null, isBundled=false)
and class 'com.kalessil.phpStorm.phpInspectionsEA.inspectors.apiUsage.UselessUnsetInspector' in PluginDescriptor(name=Php Inspections (EA Ultimate), id=com.kalessil.phpStorm.phpInspectionsUltimate-mp, descriptorPath=plugin.xml, path=~/.local/share/JetBrains/Toolbox/apps/PhpStorm/ch-0/221.5080.224.plugins/PhpInspectionsUltimate-mp.jar, version=2021.5, package=null, isBundled=false)
conflict

com.intellij.diagnostic.PluginException: Short name 'MisorderedModifiersInspection' is not unique
class 'com.kalessil.phpStorm.phpInspectionsEA.inspectors.codeStyle.MisorderedModifiersInspector' in PluginDescriptor(name=Php Inspections (EA Extended), id=com.kalessil.phpStorm.phpInspectionsEA, descriptorPath=plugin.xml, path=~/.local/share/JetBrains/Toolbox/apps/PhpStorm/ch-0/221.5080.224.plugins/PhpInspectionsEA.jar, version=4.0.7.1, package=null, isBundled=false)
and class 'com.kalessil.phpStorm.phpInspectionsEA.inspectors.codeStyle.MisorderedModifiersInspector' in PluginDescriptor(name=Php Inspections (EA Ultimate), id=com.kalessil.phpStorm.phpInspectionsUltimate-mp, descriptorPath=plugin.xml, path=~/.local/share/JetBrains/Toolbox/apps/PhpStorm/ch-0/221.5080.224.plugins/PhpInspectionsUltimate-mp.jar, version=2021.5, package=null, isBundled=false)
conflict


com.intellij.diagnostic.PluginException: Short name 'UnsupportedStringOffsetOperationsInspection' is not unique
class 'com.kalessil.phpStorm.phpInspectionsEA.inspectors.languageConstructions.UnsupportedStringOffsetOperationsInspector' in PluginDescriptor(name=Php Inspections (EA Extended), id=com.kalessil.phpStorm.phpInspectionsEA, descriptorPath=plugin.xml, path=~/.local/share/JetBrains/Toolbox/apps/PhpStorm/ch-0/221.5080.224.plugins/PhpInspectionsEA.jar, version=4.0.7.1, package=null, isBundled=false)
and class 'com.kalessil.phpStorm.phpInspectionsEA.inspectors.languageConstructions.UnsupportedStringOffsetOperationsInspector' in PluginDescriptor(name=Php Inspections (EA Ultimate), id=com.kalessil.phpStorm.phpInspectionsUltimate-mp, descriptorPath=plugin.xml, path=~/.local/share/JetBrains/Toolbox/apps/PhpStorm/ch-0/221.5080.224.plugins/PhpInspectionsUltimate-mp.jar, version=2021.5, package=null, isBundled=false)
conflict

java.lang.IllegalStateException: You must not register toolwindow programmatically so early. Rework code or use ToolWindowManager.invokeLater
```

**Comment by PaulHHM** [comment]
```
java.lang.IllegalStateException: You must not register toolwindow programmatically so early. Rework code or use ToolWindowManager.invokeLater
	at com.intellij.openapi.wm.impl.ToolWindowManagerImpl.getToolWindowPaneIfInitialized(ToolWindowManagerImpl.kt:490)
	at com.intellij.openapi.wm.impl.ToolWindowManagerImpl.registerToolWindow(ToolWindowManagerImpl.kt:928)
	at com.jetbrains.rdserver.toolWindow.BackendServerToolWindowManager.registerToolWindow(BackendServerToolWindowManager.kt:89)
	at com.intellij.openapi.wm.ToolWindowManager.registerToolWindow(ToolWindowManager.kt:37)
	at com.kalessil.phpStorm.phpInspectionsEA.EAUltimateSidebarComponent.projectOpened(EAUltimateSidebarComponent.java:48)
	at com.intellij.openapi.project.impl.ProjectManagerExImplKt$openProject$$inlined$executeInEdtWithProgress$1$1$lambda$1.invoke(ProjectManagerExImpl.kt:503)
	at com.intellij.openapi.project.impl.ProjectManagerExImplKt$openProject$$inlined$executeInEdtWithProgress$1$1$lambda$1.invoke(ProjectManagerExImpl.kt)
	at com.intellij.serviceContainer.ComponentManagerImpl.processInitializedComponents(ComponentManagerImpl.kt:1324)
	at com.intellij.openapi.project.impl.ProjectManagerExImplKt$openProject$$inlined$executeInEdtWithProgress$1$1.run(ProjectManagerExImpl.kt:611)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$12(CoreProgressManager.java:608)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:683)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:639)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:607)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.openapi.project.impl.ProjectManagerExImplKt$openProject$$inlined$executeInEdtWithProgress$1.run(ProjectManagerExImpl.kt:460)
	at com.intellij.openapi.application.TransactionGuardImpl.runWithWritingAllowed(TransactionGuardImpl.java:215)
	at com.intellij.openapi.application.TransactionGuardImpl.access$100(TransactionGuardImpl.java:22)
	at com.intellij.openapi.application.TransactionGuardImpl$1.run(TransactionGuardImpl.java:197)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:873)
	at com.intellij.openapi.application.impl.ApplicationImpl$3.run(ApplicationImpl.java:511)
	at com.intellij.openapi.application.impl.LaterInvocator$1.run(LaterInvocator.java:96)
	at com.intellij.openapi.application.impl.FlushQueue.doRun(FlushQueue.java:69)
	at com.intellij.openapi.application.impl.FlushQueue.runNextEvent(FlushQueue.java:112)
	at com.intellij.openapi.application.impl.FlushQueue.flushNow(FlushQueue.java:42)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:776)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:746)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:898)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:746)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$6(IdeEventQueue.java:439)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:803)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$7(IdeEventQueue.java:438)
	at com.intellij.openapi.application.TransactionGuardImpl.performActivity(TransactionGuardImpl.java:106)
	at com.intellij.ide.IdeEventQueue.performActivity(IdeEventQueue.java:604)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:436)
	at com.intellij.openapi.application.impl.ApplicationImpl.runIntendedWriteActionOnCurrentThread(ApplicationImpl.java:873)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:484)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)

```

**Comment by kissifrot** [comment]
Duplicate of #1805

**Comment by Kuaile1234** [comment]
<img width="1789" alt="image" src="https://user-images.githubusercontent.com/47730112/175861178-a93dcf71-249f-4c86-b7ef-b4314805cd86.png">

**Comment by ea-inspections-team** [comment]
Thank you for reporting @izac1, all know crashed are solved in v2023.1 and 2024.1.

@izac1 @PaulHHM, @kissifrot: we are so sorry for the frustration caused by the crashes and lack of support by previous maintainers team. Please contact us on ea.inspections.team@gmail.com (we will arrange one year free license for you).
